### DISCUSSION TITLE
Cannot use import statement outside a module

### DISCUSSION THREAD

**Discussion by GuyoST** [remix-run/remix#4309] [Q&A] [answered]
Hello every one,

I'm very new to Remix.
Currently I have an express server with node, that is used as an API server only for a client app made with create-react-app.
It's working fine.

Now I need another app that must load faster that the client app, so I discovered remix and I read that it can be started from my express server.

I have 2 questions : 

### 1. more a problem than a question :

I ran `npx create-remix@latest` and selected `express` as deployment target and also selected `typescript`.

The project was created but it was a `server.js` file, but I expected a typescript file like `server.ts`. So I renamed it and added the stuff my current node server has, and I also added import like `import fs from "fs"` or `import path from "path"`.
When I compile, that's fine, but when I run it with `npm run dev` I get  👍 

```
import path from "path";
^^^^^^
SyntaxError: Cannot use import statement outside a module
```

so I tried to add "type":"module" in `package.json` but I got : 

```
Building Remix app in production mode...
Error: Error loading Remix config at C:\...path...\remix.config.js
ReferenceError: module is not defined in ES module scope
This file is being treated as an ES module because it has a '.js' file extension and 'C:\...path...\remix\package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.
```

So I renamed to `server.cts` and I got the first error again.

Is there a way to use import ? Because all my current server uses import statement, so that would be very unconvenient to change all the files.

### Question 2 : Distinguished the routes and request per domain name

The create-react-app is for the admin area of the application. It uses the server as an API server. For example to get customers list it will call : https://admin-server.domain.com/customers

And let's say that the admin area is accessible from https://admin.domain.com

Then the others website that I want to render server side with remix will be accessible from other domains, like client.domain.com for example.

Is it possible to say the server to "ignore" all the requests defined in express if the domain of the request is not admin.domain.com ?
So if a route in remix is /customers, it should not be confused with /customers used by the request for the api.

My idea was to start 2 express app at same time on 2 different ports (one for api, one for remix app), and admin website will use one and clients website will use the other one, but if there is a better practice for that, let me know 🙂

Thank you for your help 🙂

**Comment by GuyoST** [comment, marked as answer]
Hello,

I just found some information about my questions, so I post it here in case some other people are interested.

For 1 : I found this : https://github.com/remix-run/remix/discussions/3244
So according to that document, I think there will be soon the availability to use `npm run dev` with a custom node JS when the `serverEntryFile `will be available in a next release ?

For 2 : Finally I will not use it, I will keep my API server as it is, and I will use the remix server as a client of it.

**Reply by kiliman** [reply]
Here's the PR for `serverEntryFile` https://github.com/remix-run/remix/pull/4123 Not sure when or if it'll get merged.
